tests/evm/test_return_revert.py

```diff
@@ -110,8 +112,171 @@ def test_is_root_not_create(
     )
 
 
-# TODO: test_return_is_root_is_create
-# TODO: test_return_not_root_is_create
+TESTING_DATA_IS_CREATE = (
+    (Transaction(), 1_000_000, True, True, 4, 10),  # RETURN, ROOT
+    (Transaction(), 1_000_000, True, False, 4, 10),  # REVERT, ROOT
+    (Transaction(), 1_000_000, False, True, 4, 10),  # RETURN, not ROOT
+    (Transaction(), 1_000_000, False, False, 4, 10),  # REVERT, not ROOT
+)
+
+
+@pytest.mark.parametrize(
+    "tx, gas_available, is_root, is_return, return_offset, return_length",
+    TESTING_DATA_IS_CREATE,
+)
+def test_is_create(
+    tx: Transaction,
+    gas_available: int,
+    is_root: bool,
+    is_return: bool,
+    return_offset: int,
+    return_length: int,
+):
+    randomness = rand_fq()
+    CALLEE_ADDRESS = 0xFE
+
+    block = Block()
+
+    init_bytecode = gen_bytecode(is_return, return_offset, return_length)
+    deployment_bytecode_hash = RLC(init_bytecode.hash(), randomness)
+
+    # gas
+    _, gas_memory_expansion = memory_expansion(0, return_offset + return_length)
+    gas_cost = return_length * GAS_COST_CODE_DEPOSIT
+    gas_left = gas_available - gas_cost
+
+    return_offset_rlc = RLC(return_offset, randomness)
+    return_length_rlc = RLC(return_length, randomness)
+    caller_id = 1
+    rw_counter = 16 if is_root else 27
+    callee_id = rw_counter
+
+    rw_dict = (
+        RWDictionary(rw_counter)
+        .call_context_read(callee_id, CallContextFieldTag.IsSuccess, int(is_return))
+        .stack_read(callee_id, 1022, return_offset_rlc)
+        .stack_read(callee_id, 1023, return_length_rlc)
+        .call_context_read(callee_id, CallContextFieldTag.CalleeAddress, int(CALLEE_ADDRESS))
+        .account_write(
+            CALLEE_ADDRESS,
+            AccountFieldTag.CodeHash,
+            deployment_bytecode_hash,
+            RLC(EMPTY_CODE_HASH),
+        )
+    )
+
+    # copy_table
+    src_data = dict(
+        [
+            (
+                i,
+                (
+                    init_bytecode.code[i - return_offset + 1],
+                    init_bytecode.is_code[i - return_offset + 1],
+                )
+                if i - return_offset + 1 < len(init_bytecode.code)
+                else (0, 0),
+            )
+            for i in range(return_offset, return_offset + return_length)
+        ]
+    )
```

Comment:
```suggestion
    src_data = dict(
        [
            (
                i,
                (
                    init_bytecode.code[i - return_offset],
                    init_bytecode.is_code[i - return_offset],
                )
                if i - return_offset < len(init_bytecode.code)
                else (0, 0),
            )
            for i in range(return_offset, return_offset + return_length)
        ]
    )
```

Reply:
fixed in [e5acf67](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/355/commits/e5acf67005cc8571ca85bdd0006b517e39ac0cb3)